### PR TITLE
no_proxy inherits the system settings to adapt to some special scenarios

### PR DESCRIPTION
**Pull request by tanjing2020**
#### What type of PR is this?
/kind failing-test

#### What this PR does / why we need it:
```
export no_proxy=“127.0.0.1, localhost”
```
Directly assigning no_proxy will not inherit the system settings, which will cause some access errors due to incorrect use of the proxy. 
If we inherit the system settings, we can get more flexible configuration without embedded modification.
```
export no_proxy=“127.0.0.1, localhost, $no_proxy”
```
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/lib/init.sh [hunk 1]
```diff
@@ -41,7 +41,11 @@ KUBE_RSYNC_COMPRESS="${KUBE_RSYNC_COMPRESS:-0}"
 
 # Set no_proxy for localhost if behind a proxy, otherwise,
 # the connections to localhost in scripts will time out
-export no_proxy=127.0.0.1,localhost
+if [ ! -n "$no_proxy" ]; then
```
Comment (BenTheElder) [inline hack/lib/init.sh:44]: we can do this simpler like:
```bash
export no_proxy="127.0.0.1,localhost${no_proxy:+,${no_proxy}}"
```
no need for the if block.

**Review by BenTheElder — CHANGES_REQUESTED** [review]
the current form is broken when no_proxy is not set, which is why CI is failing.
see my suggestion for a simpler form that won't be.

File: hack/lib/init.sh (hunk 1 above)
Comment (tanjing2020) [reply hack/lib/init.sh]: Thank you very much for your suggestion. 
Done.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
thanks!

**Comment by pacoxu** [comment]
not sure which sig is correct.

/sig contributor-experience

**Pull request merged**